File: solutions/4000-count-bowl-subarrays/count-bowl-subarrays.cpp
class Solution {
public:
    using ll = long long;
    ll bowlSubarrays(vector<int>& nums) {
        ll res = 0;
        vector<int> stk;
        int n = nums.size();

        for (int i = 0; i < n; i++) {
            while (!stk.empty() and nums[stk.back()] < nums[i]) {
                if (i - stk.back() >= 2) {
                    res++;
                }
                stk.pop_back();
            }
            if (!stk.empty() and i - stk.back() >= 2) {
                res++;
            }
            stk.push_back(i);
        }
        return res;
    }
};

File: solutions/4000-count-bowl-subarrays/count-bowl-subarrays.py
class Solution(object):
    def bowlSubarrays(self, nums):
        res = 0
        stk = []
        n = len(nums)

        for i in range(n):
            while stk and nums[stk[-1]] < nums[i]:
                if i - stk[-1] >= 2:
                    res += 1
                stk.pop()
            if stk and i - stk[-1] >= 2:
                res += 1
            stk.append(i)
        return res